src/libstd/sys/vxworks/process/process_common.rs

```diff
@@ -90,8 +90,8 @@ impl Command {
         let program = os2c(program, &mut saw_nul);
         Command {
             argv: Argv(vec![program.as_ptr(), ptr::null()]),
+            args: vec![program.clone()],
```

Comment:
Wouldn't it be more "natural" to use `program` here and `program.clone()` on the next line?

I.e.

```
args: vec![program],
program: program.clone(),
```

**Rationale:**

argv[0] and args[0] should be related as follows:

args[0] owns a CString representing "argv[0]"
argv[0] is a raw pointer to the start of that CString

Thus args is used to manage the lifetime of the strings, and argv is used to hold pointers to be passed to OS functions.

This is consistent with what we do in set_arg_0 and arg; in both cases we put an owned string in args, and a raw pointer to that string in argv.

But the current code here puts a pointer in argv[0] but args[0] holds a completely different string that we don't care about! It will still work, because "program" owns the string we do care about, but it is a bit confusing. Do you agree?

Reply:
I am OK with both ways. The code was simply copied from the code for unix, don't we want to keep align with unix if possible?

Reply:
"don't we want to keep align with unix if possible?" Yes, but we also need to impose our own high standards for code in the VxWorks tree. Sometimes the Unix code may not be ideal.

I suppose we could also propose making the same change for Unix ...!

But if you don't think this change is necessary then I'm not going to force the issue, since the code works correctly in either case.

Reply:
I am not 100% confident to convince the community to accept the change, I would like to keep it as it is.